### PR TITLE
Rework examples and top level README

### PR DESCRIPTION
**Pull request by chris-crone**
**- What I did**
See commits for details

### CODE REVIEW

**Comment by chris-crone** [comment]
Note that the CNAB helm example does not work. It will only work once we handle credential sets as YAML files.

File: README.md [hunk 1]
```diff
@@ -1,25 +1,24 @@
-# Docker Application Packages
+# Docker Application
 
-An *experimental* utility to help make Compose files more reusable and sharable.
+A utility to help make Compose files more reusable and shareable.
 
-## CNAB support (preview)
+## The problem application packages solves
```
Comment (silvin-lubecki) [inline README.md]: Application Packages ?

File: README.md [hunk 2]
```diff
@@ -68,93 +68,132 @@ services:
...
   hello:
-    image: hashicorp/http-echo:${version}
+    image: hashicorp/http-echo
     command: ["-text", "${text}"]
     ports:
       - ${port}:5678
 ```
 
-Finally you can test everything is working, by rendering the Compose file with the provided default values.
+You can test everything is working, by inspecting the application.
```
Comment (silvin-lubecki) [inline README.md]: the Application Package?

File: README.md [hunk 3]
```diff
@@ -68,93 +68,132 @@ services:
...
     ports:
     - mode: ingress
       target: 5678
       published: 5678
       protocol: tcp
 ```
 
-You can then use that Compose file like any other. You could save it to disk or pipe it straight to `docker stack` or `docker-compose` to launch the application.
+You can then use that Compose file like any other. You could save it to disk or
+pipe it straight to `docker stack` or `docker-compose` to launch the
```
Comment (silvin-lubecki) [inline README.md]: Launch? Or deploy/run/install ?

File: README.md [hunk 4]
```diff
@@ -68,93 +68,132 @@ services:
...
 
-```
-$ docker-app render | docker-compose -f - up
+```console
+$ docker app render | docker-compose -f - up
 ```
 
-This is where it gets interesting. We can override those parameters at runtime, using the `--set` option. Let's specify different option and run `render` again:
+This is where it gets interesting. We can override those parameters at runtime,
+using the `--set` option. Let's specify different option and run `render` again:
```
Comment (silvin-lubecki) [inline README.md]: s/option/options

File: README.md [hunk 5]
```diff
@@ -68,93 +68,132 @@ services:
...
 ```
-$ docker-app render -f prod.yml
+
+You can share your application package by pushing it to a container registry.
+
+```console
+$ docker app push --tag myrepo/hello:0.1.0
+```
+
+Others can then use your application by specifying the registry tag.
```
Comment (silvin-lubecki) [inline README.md]: Application Package ?

File: README.md [hunk 6]
```diff
@@ -68,93 +68,132 @@ services:
...
 ```
 
 You can then run using that configuration file like so:
 
+```console
+$ docker app render --parameters-file prod.yml
 ```
-$ docker-app render -f prod.yml
+
+You can share your application package by pushing it to a container registry.
```
Comment (silvin-lubecki) [inline README.md]: Application Package

File: README.md (hunk 5 above)
Comment (silvin-lubecki) [inline README.md]: Eventually add that every command can work with a registry tag (render/inspect/install/...) ?

File: README.md [hunk 7]
```diff
@@ -68,93 +68,132 @@ services:
...
 More examples are available in the [examples](examples) directory.
 
 ## CNAB
 
-Under the hood `docker-app` is CNAB compliant. It generates a CNAB from your application source and is able to install and manage any other CNAB too.
-CNAB specifies three actions which `docker-app` provides as commands:
-- `install`
-- `upgrade`
-- `uninstall`
+Under the hood `docker app` is CNAB compliant. It generates a CNAB from your
```
Comment (silvin-lubecki) [inline README.md]: Link CNAB to cnab specifications or cnab.io?

File: README.md [hunk 8]
```diff
@@ -68,93 +68,132 @@ services:
...
+application source and is able to install and manage any other CNAB too. CNAB
+specifies three actions which `docker app` provides as commands:
+* `install`
+* `upgrade`
+* `uninstall`
 
-**Note**: These commands need a Docker Context so that `docker-app` knows which endpoint and orchestrator to target.
+**Note**: These commands need a Docker Context so that `docker app` knows which
+Docker endpoint to target.
```
Comment (silvin-lubecki) [inline README.md]: What about adding some material explaining the difference between the context to execute the invocation image and the target context?

File: examples_test.go [hunk 9]
```diff
@@ -1,48 +1,45 @@
...
-		"text": "hello examples!",
-	}, nil)
-	if err != nil {
-		panic("cannot render application")
-	}
-	// Marshal it to yaml (to display it)
-	s, err := yaml.Marshal(c)
-	if err != nil {
-		panic("cannot marshall the composefile in yaml")
+var examplePaths = map[string]types.AppSourceKind{
```
Comment (silvin-lubecki) [inline examples_test.go]: This `Example` function is not what you think it is. There are already e2e tests checking that examples are still up-to-date: https://github.com/docker/app/blob/master/e2e/example_test.go#L13
An Example function is like a special test function, but only checking the stdout with the comment `// Output...`.
It also appears in the godoc.
I think you can safely trash this whole file 👍

File: README.md (hunk 2 above)
Comment (simonferquel) [reply README.md]: or definition ?

File: README.md (hunk 8 above)
Comment (simonferquel) [reply README.md]: Yep, and maybe state that in the future, we'll probably support `kubeconfig` files for kubernetes workloads as well

File: README.md [hunk 10]
```diff
@@ -68,93 +68,132 @@ services:
...
 
 ```console
 $ docker context create swarm --description "swarm context" --default-stack-orchestrator=swarm --docker-host=unix:///var/run/docker.sock
```
Comment (simonferquel) [inline README.md]:
```suggestion
$ docker context create swarm --description "swarm context" --default-stack-orchestrator=swarm --docker host=unix:///var/run/docker.sock
```

File: README.md (hunk 8 above)
Comment (simonferquel) [reply README.md]: That is not strictly true anymore: default target context is the same as current active context.
Defining a target context is only required for installing on a remote cluester

File: README.md [hunk 11]
```diff
@@ -238,128 +264,131 @@ Invoke-WebRequest -Uri https://github.com/docker/app/releases/download/v0.6.0/do
...
+We have lots of ideas for making Compose-based applications easier to share and
+reuse, and making applications a first-class part of the Docker toolchain.
+Please let us know what you think about this initial release and about any of
+the ideas below:
 
 * Introducing environments to the parameters file
 * Docker images which launch the application when run
 * Built-in commands for running applications
 * Saving required images into the application artifact to support offline installation
 * Signing applications with notary
```
Comment (simonferquel) [inline README.md]: Are those topics still up to date ?

File: README.md (hunk 2 above)
Comment (chris-crone) [reply README.md]: We're using "definition" or "source" for the on disk representation and "package" for the image format.

File: README.md (hunk 8 above)
Comment (chris-crone) [reply README.md]: I actually think that this shouldn't be under the CNAB section. I'll remove it.

Comment (chris-crone) [reply README.md]: The context part should be in another example. We don't need it when we're showing CNAB compliance.

File: README.md (hunk 11 above)
Comment (chris-crone) [reply README.md]: Nope. I'll remove them until we have a more clear roadmap to share.

**Comment by chris-crone** [comment]
@simonferquel @silvin-lubecki Requested changes are in two latest commits. I'll squash the README ones before merging.

**Comment by chris-crone** [comment]
Rebased after merge of #509

**Pull request merged**